app/src/androidTest/java/ch/epfl/cs311/wanderwave/model/BeaconConnectionTest.kt

```diff
@@ -23,137 +23,70 @@ public class BeaconConnectionTest {
-      val beacon =
-          Beacon(
-              id = "testBeacon",
-              location = Location(1.0, 1.0, "Test Location"),
-              tracks = listOf(Track("testTrack", "Test Title", "Test Artist")))
-
-      beaconConnection.addItemWithId(beacon)
-
-      val retrievedBeacon = beaconConnection.getItem("testBeacon").first()
-      Log.d("Firestore", "$retrievedBeacon $beacon")
-      assert(beacon == retrievedBeacon)
-    }
-  }
-
-  @Test
-  fun testAddItem() = runBlocking {
     // Mock data
-    val beacon =
+    beacon =
         Beacon(
             id = "testBeacon",
             location = Location(1.0, 1.0, "Test Location"),
             tracks = listOf(Track("testTrack", "Test Title", "Test Artist")))
 
+    // Define behavior for the mocks
+    every { firestore.collection(any()) } returns mockk(relaxed = true)
+    every { collectionReference.document(beacon.id) } returns documentReference
+    every { firestore.collection(any()) } returns collectionReference
+
+    // Pass the mock Firestore instance to your BeaconConnection
+    beaconConnection = BeaconConnection(firestore)
+  }
+
+  @Test
+  fun testAddItem() {
     // Call the function under test
     beaconConnection.addItem(beacon)
 
-    // No verification is needed for interactions with the real object
+    // Verify that either the set function is called
+    verify { collectionReference.add(any()) }
   }
 
   @Test
   fun testUpdateItem() = runBlocking {
-    // Mock data
-    val beacon =
-        Beacon(
-            id = "testBeacon",
-            location = Location(1.0, 1.0, "Test Location"),
-            tracks = listOf(Track("testTrack", "Test Title", "Test Artist")))
-
     // Call the function under test
     beaconConnection.updateItem(beacon)
 
-    // No verification is needed for interactions with the real object
+    // Verify that the set function is called on the document with the correct id
+    verify { documentReference.set(any()) }
   }
 
-  // If someone knows how to deal with the flow already being null and how to test it, please let me
-  // know
   @Test
-  fun testGetNonexistentItem() = runBlocking {
-    // By default the test passes, if a value is emitted the test fails
-    withTimeout(20000) {
-      // Flag to indicate if the flow emits any value
-      var valueEmitted = false
-
-      // Collect the flow within a 2-second timeout
-      measureTimeMillis {
-        withTimeoutOrNull(2000) {
-          beaconConnection.getItem("nonexistentBeacon").collect {
-            valueEmitted = true // Set the flag if the flow emits any value
-          }
-        }
-      }
-
-      // Assert that the flow didn't emit anything within the timeout
-      assert(valueEmitted.not()) { "Flow emitted unexpected value" }
-    }
-  }
+  fun testGetItem() = runBlocking {
+    withTimeout(3000) {
+      // Call the function under test
+      val retrievedBeacon = beaconConnection.getItem("testBeacon")
```

Comment:
This just gets the flow, but we need to also consume it to check that it actually creates a value, eg with a `.first()`. When trying that locally, it doesn't pass yet. This would also help increase coverage.

Reply:
I have tried to make the test better but it's very complicated to make the get item of the beacon work, because it needs to fetch the tracks after it has already fetched the beacon, so it's quite hard to make it work with the mocking we have. It seems like mocking a document like that has some limitations.